core/src/main/java/com/graphhopper/storage/TurnCostExtension.java

```diff
@@ -197,6 +197,10 @@ public long getTurnCostFlags(int edgeFrom, int nodeVia, int edgeTo) {
         return nextCostFlags(edgeFrom, nodeVia, edgeTo);
     }
 
+    public boolean isUTurn(int edgeFrom, int edgeTo) {
+        return edgeFrom == edgeTo;
+    }
```

Comment:
Wouldn't it make more sense in the TurnWeighting as you wrote in #1593?

Reply:
Yes I was tempted to do it, especially since `AbstractBidirectionEdgeCHNoSOD` already has a reference to a `TurnWeighting`. Then again to make it really consistent we should also use `isUTurn` in `AbstractRoutingAlgorithm#accept` (but I haven't), but this one only has a reference to `Weighting` (while it could obtain a `TurnCostExtension` reference from `Graph.getExtension`).  In any case `TurnCostExtension` needs this method, so we can override it in `QueryGraphTurnExt`. The routing algorithms could call it via something like `TurnWeighting#isUTurn`. I don't know. Either way is fine for me. Maybe postpone this decision until doing a proper u-turn clean up in #1520 ?

Reply:
Thanks for the explanation - makes sense

> Maybe postpone this decision until doing a proper u-turn clean up in #1520 ?

:+1: